packages/contentful-app-extensions/contributing-cohort-memberships/src/test/locations/Field.test.tsx

```diff
@@ -0,0 +1,266 @@
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(useEntity).toHaveBeenCalledWith(
+          'Entry',
+          'contributing-cohort-1',
+        );
+        expect(screen.getByText('My Contributing Cohort')).toBeInTheDocument();
+        expect(screen.getByText('Lead')).toBeInTheDocument();
+      });
+    });
+
+    it('displays an error message if no contributing cohort is selected', async () => {
+      const props = {
+        entity: {
+          fields: {
+            role: {
+              'en-US': 'Project Manager',
+            },
+            contributingCohort: null,
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(
+          screen.getByText('No contributing cohort selected'),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it('calls the `onEdit` handler in contentful to open the related entity when clicked', async () => {
+      const props = {
+        entity: {
+          fields: {
+            role: null,
+            contributingCohort: null,
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(
+          screen.getByText('No contributing cohort selected'),
+        ).toBeInTheDocument();
+      });
+      fireEvent.click(screen.getByText('No contributing cohort selected'));
+      await waitFor(() => {
+        expect(props.onEdit).toHaveBeenCalled();
+      });
+    });
+
+    it('renders a remove button in the card menu', async () => {
+      const props = {
+        entity: {
+          fields: {
+            role: null,
+            contributingCohort: null,
+          },
+          sys: {
+            type: 'Entry',
+            publishedVersion: 1,
+            version: 1,
+          },
+        },
+        onEdit: jest.fn(),
+        onRemove: jest.fn(),
+      } as unknown as CustomEntityCardProps;
+
+      render(<CustomCard {...props} />);
+
+      await waitFor(() => {
+        expect(
+          screen.getByText('No contributing cohort selected'),
+        ).toBeInTheDocument();
+      });
+      fireEvent.click(screen.getByLabelText('Actions'));
+      await waitFor(() => {
+        expect(screen.getByText('Remove')).toBeVisible();
+      });
```

Comment:
we should be able to use findByText (and prefer to use findByRole)

for example:

```
expect(await screen.findByRole('button', { name: /remove/i })).toBeVisible();
```